circus-train-s3-s3-copier/src/main/java/com/hotels/bdp/circustrain/s3s3copier/aws/JceksAmazonS3ClientFactory.java

```diff
@@ -162,15 +147,17 @@ private AmazonS3 buildClient(String region, S3S3CopierOptions s3s3CopierOptions,
     return builder.build();
   }
 
-  private HadoopAWSCredentialProviderChain getCredentialsProviderChain() {
-    if (security.getCredentialProvider() == null) {
-      return new HadoopAWSCredentialProviderChain();
-    }
-    return new HadoopAWSCredentialProviderChain(security.getCredentialProvider());
-  }
+  private HadoopAWSCredentialProviderChain getCredentialsProviderChain(
+      S3S3CopierOptions s3s3CopierOptions,
+      boolean assumedRoleProvided) {
 
-  private HadoopAWSCredentialProviderChain getCredentialsProviderChain(Configuration conf) {
-    return new HadoopAWSCredentialProviderChain(conf);
+    if (assumedRoleProvided) {
+      setConfRole(conf, s3s3CopierOptions.getAssumedRole());
```

Comment:
This setConfRole method should be `createNewConf(conf, s3s3CopierOptions.getAssumedRole())` and return a copy of the conf as we are changing it and the conf might be used elsewhere we don't side effects I think.